### ISSUE TITLE
[stepfunctions] `nextToken` should not be required in output type

### ISSUE THREAD

**Issue by taesungh** [youtype/mypy_boto3_builder#318]
Similar to #238, the type definitions for the responses of some of the methods on the [AWS Step Functions client](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/stepfunctions.html) specify `nextToken` as required in the `TypedDict` when it is not always returned.

For example, here is `GetExecutionHistoryOutputTypeDef` (for [`get_execution_history`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/stepfunctions/client/get_execution_history.html))
```python
GetExecutionHistoryOutputTypeDef = TypedDict(
    "GetExecutionHistoryOutputTypeDef",
    {
        "events": List["HistoryEventTypeDef"],
        "nextToken": str,
        "ResponseMetadata": "ResponseMetadataTypeDef",
    },
)
```
and this also applies to `ListExecutionsOutputTypeDef` and a few other output types

#### Example query and response
There may be multiple pages for execution history,
```python
import boto3
sfn_client = boto3.client("stepfunctions")
history = sfn_client.get_execution_history(executionArn="arn:...")
```
but the response doesn't include `nextToken` when there are no more pages.
```python
{
    "events": [
        { ... },
    ],
    "ResponseMetadata": { ... },
}
```

#### Additional context
Referencing [mypy-boto3-stepfunctions v1.35.9](https://pypi.org/project/mypy-boto3-stepfunctions/1.35.9/)

**Comment by vemel** [comment]
Hello! Thank you for the report.

I already have a list of keys that should not be marked as required in output shapes. I can mark `nextToken` as `Optional` as well. I will check how many services are affected.

As far as I know, `botocore` team tries to follow the same pagination pattern for all services, but we will see.

**Comment by vemel** [comment]
Actually, as I see `NextToken` is already marked as `NotRequired`, but `nextToken` is not :) So, not all services follow the same pagination pattern.

**Comment by vemel** [comment]
Fixed in `mypy-boto3-builder 8.1.3`. Also, I released `mypy-boto3-stepfunctions 1.35.46` with the fix included. Please update and let me know if it works as expected.

**Comment by taesungh** [comment]
Thanks for the quick turnaround, this looks to be correct now in the latest release.
```python
GetExecutionHistoryOutputTypeDef = TypedDict(
    "GetExecutionHistoryOutputTypeDef",
    {
        "events": List[HistoryEventTypeDef],
        "ResponseMetadata": ResponseMetadataTypeDef,
        "nextToken": NotRequired[str],
    },
)
```
